src/apic-extension/azext_apic_extension/command_patches.py

```diff
@@ -519,3 +543,199 @@ def pre_operations(self):
+        # pylint: disable=protected-access
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        # Remove the azure-api-management-source parameter
+        args_schema.azure_api_management_source._registered = False
+        # Remove the amazon-api-gateway-source parameter
+        args_schema.amazon_api_gateway_source._registered = False
+        # Remove the api_source_type parameter, will set it for users in pre_operations
+        args_schema.api_source_type._required = False
+        args_schema.api_source_type._registered = False
+
+        # Create arg group for AzureApiManagementSource
+
+        args_schema.msi_resource_id = AAZResourceIdArg(
+            options=["--msi-resource-id"],
+            arg_group="AzureApiManagementSource",
+            help="The resource ID of the managed identity that has access to the API Management instance.",
+        )
+
+        args_schema.azure_apim = AAZStrArg(
+            options=["--azure-apim"],
+            arg_group="AzureApiManagementSource",
+            help="The name or resource id of the source APIM instance.",
+            required=True
+        )
+
+        return args_schema
+
+    def pre_operations(self):
+        # Set apim_resource_id based on user input
+        super().pre_operations()
+        args = self.ctx.args
+
+        if not is_valid_resource_id(args.azure_apim.to_serialized_data()):
+            # The APIM is in the same resource group
+            resource_group = args.resource_group
+            subscription_id = self.ctx.subscription_id
+            apim_resource_id = (f"/subscriptions/{subscription_id}/resourceGroups/{resource_group}/providers/"
+                                f"Microsoft.ApiManagement/service/{args.azure_apim}")
+        else:
+            apim_resource_id = args.azure_apim
+
+        args.azure_api_management_source = {
+            "msi_resource_id": args.msi_resource_id,
+            "apim_resource_id": apim_resource_id
+        }
+
+        # Set api_source_type
+        args.api_source_type = "AzureApiManagement"
+
+
+# `az apic api-analysis` commands
+class CreateApiAnalysisConfig(DefaultWorkspaceParameter, CreateApiAnalysis):
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        # pylint: disable=protected-access
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.filter._registered = False
+        return args_schema
+
+    def pre_operations(self):
+        super().pre_operations()
+        args = self.ctx.args
+        args.analyzer_type = "spectral"
+
+
+class DeleteApiAnalysisConfig(DefaultWorkspaceParameter, DeleteApiAnalysis):
+    pass
+
+
+class ImportApiAnalysisRuleset(DefaultWorkspaceParameter, ImportRuleset):
+    # pylint: disable=C0301
+    # Zip and encode the ruleset folder to base64
+    def zip_folder_to_buffer(self, folder_path):
+        # pylint: disable=unused-variable
```

Comment:
You could try  to add `_` in front of the unused variables to avoid the lint warning, which seems widely used in many projects when some function parameters or returned value from external library is indeed not used.